### PR TITLE
ARROW-9629: [Python] Fix kartothek integration tests by fixing dependencies

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
cc @xhochy 

Not a very robust fix (future-wise) in case other test dependencies get added, but not sure if there is a better way (we could in principle use their `conda-test-requirements.txt` file), and this at least fix it I think.

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-kartothek-latest test-conda-python-3.7-kartothek-master

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-kartothek-latest test-conda-python-3.7-kartothek-master

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-kartothek-latest test-conda-python-3.7-kartothek-master

**Comment by jorisvandenbossche** [comment]
The tests now run, but there are some actual failures now: see eg https://github.com/ursa-labs/crossbow/runs/941860497?check_suite_focus=true

File: ci/docker/conda-python-kartothek.dockerfile [hunk 1]
```diff
@@ -32,7 +32,14 @@ RUN conda install -c conda-forge -q \
         storefact \
         toolz \
         urlquote \
-        zstandard && \
+        zstandard \
+        attrs \
```
Comment (pitrou) [inline ci/docker/conda-python-kartothek.dockerfile:36]: Can we keep those deps in alphabetical order?

**Review by pitrou — APPROVED** [review]
LGTM

File: ci/docker/conda-python-kartothek.dockerfile (hunk 1 above)
Comment (jorisvandenbossche) [reply ci/docker/conda-python-kartothek.dockerfile]: Updated

**Comment by jorisvandenbossche** [comment]
cc @fjetter @xhochy in case if are familiar with those failures

There are several failures like this (related to exit codes):

```
_________________________________ test_simple __________________________________

cli = <function cli.<locals>._cli at 0x7ff27b5fed40>
built_cube = Cube(dimension_columns=('x', 'y'), partition_columns=('p', 'q'), uuid_prefix='my_cube', seed_dataset='source', index_columns=frozenset({'i2', 'i1'}))
mock_prompt = <function mock_prompt.<locals>._f at 0x7ff27b5b6830>
df_complete =       i1         i2  p  ...                                                 v2  x  y
0   True 2018-01-01  0  ...      ... 8, 9]  1  0
3  False 2019-01-01  0  ...  [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,...  1  1

[4 rows x 9 columns]
skv = None

    def test_simple(cli, built_cube, mock_prompt, df_complete, skv):
        mock_prompt(["", ""])  # conditions  # payload
        result = cli("--store=cubes", "my_cube", "query", input="df\n")
    
        str_df = str(df_complete.loc[:, ["p", "q", "x", "y"]].reset_index(drop=True))
    
>       assert result.exit_code == 1
E       assert 2 == 1
E        +  where 2 = <Result SystemExit(2)>.exit_code

/kartothek/tests/cli/test_query.py:64: AssertionError
```

and there are a few failures like

```
________ test_predicate_pushdown_null_col[True-serialiser4-df3-value3] _________

store = <conftest.FakeStore object at 0x7ff242feef50>, df =     t
0 NaT
value = Timestamp('2017-01-01 00:00:00'), predicate_pushdown_to_io = True
serialiser = CsvSerializer(compress=True)

    @pytest.mark.parametrize(
        "df,value",
        [
            (pd.DataFrame({"u": pd.Series([None], dtype=object)}), "foo"),
            (pd.DataFrame({"b": pd.Series([None], dtype=object)}), b"foo"),
            (pd.DataFrame({"f": pd.Series([np.nan], dtype=float)}), 1.2),
            (
                pd.DataFrame({"t": pd.Series([pd.NaT], dtype="datetime64[ns]")}),
                pd.Timestamp("2017"),
            ),
        ],
    )
    @predicate_serialisers
    @pytest.mark.parametrize("predicate_pushdown_to_io", [True, False])
    def test_predicate_pushdown_null_col(
        store, df, value, predicate_pushdown_to_io, serialiser
    ):
        key = serialiser.store(store, "prefix", df)
        expected = df.iloc[[]].copy()
        predicates = [[(df.columns[0], "==", value)]]
        result = serialiser.restore_dataframe(
            store,
            key,
            predicate_pushdown_to_io=predicate_pushdown_to_io,
            predicates=predicates,
        )
        pdt.assert_frame_equal(
            result.reset_index(drop=True),
            expected.reset_index(drop=True),
>           check_dtype=serialiser.type_stable,
        )

tests/serialization/test_dataframe.py:529: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

left = array([], dtype=float64)
right = <DatetimeArray>
[]
Length: 0, dtype: datetime64[ns], check_dtype = True
index_values = array([], dtype=int64)
check_less_precise = <object object at 0x7ff2986040f0>, check_exact = False
rtol = 1e-05, atol = 1e-08

>       assert isinstance(left, ExtensionArray), "left is not an ExtensionArray"
E       AssertionError: left is not an ExtensionArray

/opt/conda/envs/arrow/lib/python3.7/site-packages/pandas/_testing.py:1175: AssertionError
```

Since this is related to a pandas testing utility, does kartothek already support pandas 1.1?

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-kartothek-latest test-conda-python-3.7-kartothek-master

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-kartothek-latest test-conda-python-3.7-kartothek-master

**Comment by fjetter** [comment]
@jorisvandenbossche We've had a few issues with pandas 1.1.0. I'll ping again once we have a compatible release. Thanks for looking into this!

xref
https://github.com/JDASoftwareGroup/kartothek/pull/328
https://github.com/JDASoftwareGroup/kartothek/pull/332

**Comment by jorisvandenbossche** [comment]
@fjetter Thanks!

**Comment by fjetter** [comment]
@jorisvandenbossche Kartothek 3.14.0 is out and should be compatible with pandas and arrow now.

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-kartothek-latest test-conda-python-3.7-kartothek-master

**Comment by jorisvandenbossche** [comment]
The cli exit code failures are still present (see eg the first test output from my comment above https://github.com/apache/arrow/pull/7891#issuecomment-670478968)

I couldn't reproduce this locally, it might be that the CLI is not properly working in the docker environment on CI?

**Comment by fjetter** [comment]
Ok, at least the failures connected to extensionarrays are gone. I'll try to reproduce these failures or figure out another way to deal with this. The failing tests are all connected to a CLI and I don't think this is relevant for this integration tests.

**Comment by jorisvandenbossche** [comment]
> The failing tests are all connected to a CLI and I don't think this is relevant for this integration tests.

Yes, I was also thinking to simply skip those tests here

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-kartothek-latest test-conda-python-3.7-kartothek-master

**Review by fjetter — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Ah, green now ;)

**Pull request closed without merging**